addons/crm/tests/test_crm_lead.py

```diff
@@ -1009,3 +1011,233 @@ def test_crm_lead_mail_tracking_duration_batch(self):
+                        message_type='comment',
+                    )
+                msg = set_won_get_message(leads[3], user1)
+                self.assertEqual(msg, 'Phew, that took some effort — but you nailed it. Good job!')
+
+            with self.subTest('Fifth deal won today'):
+                msg = set_won_get_message(leads[4], user1)
+                self.assertEqual(msg, 'You\'re on fire! Fifth deal won today 🔥')
+
+            with self.subTest('First deal completed for another user'):
+                leads[5].expected_revenue = 100
+                msg = set_won_get_message(leads[5], user2)
+                self.assertEqual(msg, 'Go, go, go! Congrats for your first deal.')
+
+            with self.subTest('Closing a deal from source that already has a closed deal should not trigger \
+                the message again, even for another user in the same team'):
+                leads[6].source_id = custom_source.id
+                msg = set_won_get_message(leads[6], user2)
+                self.assertFalse(msg)
+
+            with self.subTest('Closing a deal from a country that already has a closed deal should not trigger \
+                the message again, even for another user in the same team'):
+                leads[7].country_id = self.env.ref('base.au')
+                msg = set_won_get_message(leads[7], user2)
+                self.assertFalse
+
+            new_leads = self.env['crm.lead'].create([
+                {
+                    'name': 'lead',
+                    'type': 'opportunity',
+                    'stage_id': stage1.id,
+                    'user_id': user1.id,
+                }, {
+                    'name': 'lead',
+                    'type': 'opportunity',
+                    'stage_id': stage1.id,
+                    'user_id': user1.id
+                },
+            ])
+            new_lead = new_leads[0]
+            new_lead2 = new_leads[1]
+            tracking_array += [(new_lead, defaultdict(lambda: 0)), (new_lead2, defaultdict(lambda: 0))]
+            all_leads |= new_lead | new_lead2
+
+        with self.mock_datetime_and_now(jan2):
+            interval = (jan2 - jan1_12pm).total_seconds() / 60
+            self.flush_tracking()
+            self._update_duration_tracking(tracking_array, interval)
+            all_leads._compute_duration_tracking()
+
+            with self.subTest('Fastest close in a 30-day window'):
+                msg = set_won_get_message(new_lead, user1)
+                self.assertEqual(msg, 'Wow, that was fast. That deal didn’t stand a chance!')
+
+        with self.mock_datetime_and_now(jan3_12pm):
+            interval = (jan3_12pm - jan2).total_seconds() / 60
+            self._update_duration_tracking(tracking_array, interval)
+
+            with self.subTest('Three-day streak'):
+                msg = set_won_get_message(leads[8], user1)
+                self.assertEqual(msg, 'You\'re on a winning streak. 3 deals in 3 days, congrats!')
```

Comment:
That echoes my comment from earlier as "user1" closed way more than "3 deals in 3 days" at this point.

Maybe this is functionally what we want, but that feels very strange to me, doesn't make much sense.